Mathlib/Algebra/Order/Interval.lean

```diff
@@ -304,7 +304,7 @@ namespace NonemptyInterval
 
 @[to_additive]
 theorem coe_pow_interval [OrderedCommMonoid α] (s : NonemptyInterval α) (n : ℕ) :
-    (s ^ n : Interval α) = (s : Interval α) ^ n :=
```

Comment:
@fpvandoorn: my opinion is that this version is very confusing, it reads "somewhere within these parentheses there is a coercion, I hope you can guess where". In order to know where the coercion ends up, you have to know the details of the `^` elaborator. The new version does not have this property.

Reply:
That makes sense. I personally read the (_ : _) themselves as coercions (and place them on the smallest subterm that has that type), but it is indeed not guaranteed that the coercion is there.

Reply:
Unfortunately I think your heuristic is bad, and you should stop using it:
```lean
import Mathlib.Algebra.Group.Defs
import Mathlib.Data.PNat.Basic

def add3 {α} [Add α] (a b c : α) : α := a + b + c

-- this is a tautology, the coercion is on the inside on both sides
example (a b c : PNat) : (add3 a b c : Nat) = add3 (a : Nat) b c := sorry
```

Reply:
That is fair, and I would indeed not write that LHS. The double type ascription is a possible alternative, but it is very verbose...
```
example (a b c : PNat) : ((add3 a b c : PNat) : Nat) = add3 (a : Nat) b c := sorry
```
`↑` can also be misleading, it doesn't necessarily insert a coercion there. But it's probably more reliable than type ascriptions.

bors d+